File: candidate-client/src/app/models/ReportSupervisorDTO.ts
import { ReportInterviewerDTO } from "./ReportInterviewerDTO";

export class ReportSupervisorDTO {

	id: Number;
	name: String;
	sum: Number;
	interviewers: ReportInterviewerDTO[];

	constructor(obj: any) {

		Object.assign(this, obj);

		let interviewers = [];
		for(let interviewer of obj.interviewers) {
			let parsedInterviewer = new ReportInterviewerDTO(interviewer);
			interviewers.push(parsedInterviewer);
		}

		this.interviewers = interviewers;

	}

}

File: candidate-api/src/main/java/com/nebu/candidate/dto/report/ReportSupervisorDTO.java
package com.nebu.candidate.dto.report;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class ReportSupervisorDTO {
	
	private long id;
	
	private String name;
	
	private double sum;
	
	private List<ReportInterviewerDTO> interviewers;

}
